main.js

```diff
@@ -383,13 +383,39 @@ function sortsch(sch){
 	sch.sort(function(a, b){if(b.d < a.d || (b.d == a.d && b.s < a.s)) return 1; return -1;});
 	return sch;
 }
+
+function calc_sleep_score(sch){
+	var sleep_score = 0;
+	for(var i=0; i < sch.length; i++){
+		if(sch[i].p != undefined){
+			sleep_score += sch[i].s;
+		}
+	}
+	return sleep_score;
+}
+
+function sort_for_sleep(){
+	for(var i=0; i<Math.min(10,schedules.length); i++){
```

Comment:
Is there any way to sort all schedules for this, without dying on 10^6 combinations? maybe using some kind of fancy data structure or something?

Reply:
Hmm, we can add an attribute something like a `schedules[i].score` and sort for using this value. What do you think about it?

Reply:
Yeah, probably that would work. Also a lot more elegant than a makeshift limited bubble sort.